### PR TITLE
[CAS-646] Proguard rules review

### PR DESCRIPTION
**Pull request by leandroBorgesFerreira**
[CAS-646](https://stream-io.atlassian.net/browse/CAS-646)

### 🎯 Goal

Reviewing proguard rules. 

### 🛠 Implementation details

Fixing proguard warnings in the compilation of our sample app release. I didn't find problems with proguard in our SDK, we don't use reflection, so our code is only accessed directly which doesn't cause problems with proguard. 

Adding 

### 🧪 Testing

Run the command: 
```
./gradlew stream-chat-android-ui-component-sample:bundleRelease
```

Adding
```
classpath 'com.android.tools:r8:3.0.65'
``` 

From here: https://issuetracker.google.com/issues/194915678?pli=1. That's temporary and should be fixed AGP 7.0.2 or some posterior version. 

### ☑️ Checklist

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] PR targets the `develop` branch
- ~[ ] Changelog is updated with client-facing changes~
- ~[ ] New code is covered by unit tests~
- ~[ ] Comparison screenshots added for visual changes~
- ~[ ] Affected documentation updated (docusaurus, tutorial, CMS (task created))~
- [x] Reviewers added

### 🎉 GIF

giphy
I didn't find a giphy that fits this PR...

### CODE REVIEW

File: stream-chat-android-ui-components-sample/proguard-rules.pro
Comment (leandroBorgesFerreira) [inline stream-chat-android-ui-components-sample/proguard-rules.pro:27]: Removing warnings. Resource: https://community.guardsquare.com/t/proguard-playground-released/454

File: stream-chat-android-client/consumer-proguard-rules.pro
Comment (zsmb13) [inline stream-chat-android-client/consumer-proguard-rules.pro:3]:
```suggestion
# Classes that are using with QuerySort can't be minified, because QuerySort uses reflection. If the
```

File: stream-chat-android-client/src/main/java/io/getstream/chat/android/client/api2/model/dto/ExtraDataDto.kt
Comment (zsmb13) [inline stream-chat-android-client/src/main/java/io/getstream/chat/android/client/api2/model/dto/ExtraDataDto.kt:3]:
```suggestion
internal interface ExtraDataDto
```

**Review by zsmb13 — APPROVED** [review]
Looks good to me - would love a second review before merging

File: stream-chat-android-client/consumer-proguard-rules.pro
Comment (bychkovdmitry) [reply stream-chat-android-client/consumer-proguard-rules.pro:3]: Ah, thanks

Comment (leandroBorgesFerreira) [reply stream-chat-android-client/consumer-proguard-rules.pro:3]: Yep, if you remove those rules, for example, you'll see that our sample app stops working as it should (if it is minified)

**Review by bychkovdmitry — APPROVED** [review]
Looks good to me. Also, checked the sample app with minifyEnabled and shrinkResources and found no apparent issues.

**Pull request merged**
